script.js

```diff
@@ -122,6 +122,50 @@ function closePopup() {
   document.getElementById('overlay-popup').style.display = 'none';
 }
 
+/* Save local data */
+const myData = [];
+let name;
+let email;
+let msg;
+
+function saveName() {
+  name = document.getElementById('name').value;
+  email = document.getElementById('email').value;
+  msg = document.getElementById('message').value;
+  myData[0] = name;
+  myData[1] = email;
+  myData[2] = msg;
+  localStorage.setItem('my_values', JSON.stringify(myData));
+}
+
+function saveEmail() {
+  name = document.getElementById('name').value;
+  email = document.getElementById('email').value;
+  msg = document.getElementById('message').value;
+  myData[0] = name;
+  myData[1] = email;
+  myData[2] = msg;
+  localStorage.setItem('my_values', JSON.stringify(myData));
+}
+function saveMsg() {
+  name = document.getElementById('name').value;
+  email = document.getElementById('email').value;
+  msg = document.getElementById('message').value;
+  myData[0] = name;
+  myData[1] = email;
+  myData[2] = msg;
+  localStorage.setItem('my_values', JSON.stringify(myData));
+}
+
+function loadLocalData() {
+  const d = localStorage.getItem('my_values');
+  const data = JSON.parse(d);
+  const [name, email, message] = data;
+  document.getElementById('name').value = name;
+  document.getElementById('email').value = email;
+  document.getElementById('message').value = message;
+}
+
```

Comment:
- [ ] [OPTIONAL] It would be nicer to create a new javascript file for this functionality. It makes your code more readable.